cinema/views.py

```diff
@@ -56,3 +83,54 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    @staticmethod
+    def _params_to_str(qs):
+        return [str(str_id) for str_id in qs.split(",")]
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            queryset = self.queryset\
+                .select_related("movie", "cinema_hall")\
+                .annotate(tickets_available=(
+                    F("cinema_hall__rows")
+                    * F("cinema_hall__seats_in_row")
+                    - Count("tickets")))
+
+        movie = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+        if movie:
+            movie_id = self._params_to_str(movie)
+            queryset = queryset.filter(movie__id__in=movie_id)
+        if date:
+            queryset = queryset.filter(show_time__exact=date)
```

Comment:
Filtering by `__exact` will only yield movies scheduled for midnight. I'd suggest using the `__date` lookup.